src/Util.ts

```diff
@@ -445,124 +445,133 @@ class Util extends BITBOXUtil {
-      let validations: any = await this.validateTxid(txids)
+      const validations: any = await this.validateTxid(txids)
       // console.log(`validations: ${JSON.stringify(validations, null, 2)}`)
 
-      // Extract the boolean result
-      validations = validations.map((x: any) => x.valid)
+      const outAry = [] // Output Array
 
       // Loop through each element and compute final validation on any that
       // returned true.
+      // Loop on utxos, not validations since txids can be the same and
+      // UTXOs are differentiated by vout value.
       for (let i = 0; i < utxos.length; i++) {
-        const thisValidation = validations[i]
         const thisUtxo = utxos[i]
         // console.log(`thisUtxo: ${JSON.stringify(thisUtxo, null, 2)}`)
 
-        // Only need to worry about validations that are still true.
-        if (thisValidation) {
-          const slpData = await this.decodeOpReturn(thisUtxo.txid)
-          // console.log(`slpData: ${JSON.stringify(slpData, null, 2)}`)
+        // const thisValidation = validations[i]
+        const thisValidation = validations.filter((x: any) => x.txid === thisUtxo.txid)
+        // console.log(
+        //   `thisValidation: ${JSON.stringify(thisValidation, null, 2)}`
+        // )
 
-          // Handle Genesis SLP transactions.
-          if (slpData.transactionType === "genesis") {
-            if (
-              thisUtxo.vout !== slpData.mintBatonVout && // UTXO is not a mint baton output.
-              thisUtxo.vout !== 1 // UTXO is not the reciever of the genesis or mint tokens.
-            ) {
-              // Can safely be marked as false.
-              validations[i] = false
-            }
-
-            // If this is a valid SLP UTXO, then return the decoded OP_RETURN data.
-            else {
-              thisUtxo.tokenType = "minting-baton"
-              thisUtxo.tokenId = thisUtxo.txid
-              thisUtxo.tokenTicker = slpData.ticker
-              thisUtxo.tokenName = slpData.name
-              thisUtxo.tokenDocumentUrl = slpData.documentUrl
-              thisUtxo.tokenDocumentHash = slpData.documentHash
-              thisUtxo.decimals = slpData.decimals
-
-              // something
-              validations[i] = thisUtxo
-            }
+        // If the utxo is not SLP, then skip the loop.
+        if (!thisValidation[0].valid) {
+          outAry[i] = false
+          continue
+        }
+
+        const slpData = await this.decodeOpReturn(thisUtxo.txid)
+        // console.log(`slpData: ${JSON.stringify(slpData, null, 2)}`)
+
+        // Handle Genesis SLP transactions.
+        if (slpData.transactionType === "genesis") {
+          if (
+            thisUtxo.vout !== slpData.mintBatonVout && // UTXO is not a mint baton output.
+            thisUtxo.vout !== 1 // UTXO is not the reciever of the genesis or mint tokens.
+          ) {
+            // Can safely be marked as false.
+            outAry[i] = false
```

Comment:
These are the main changes in this PR. They center around more careful handling of the arrays and their elements.